tests/everflow/everflow_test_utilities.py

```diff
@@ -247,18 +291,62 @@ def get_port_info(in_port_list, out_port_list, out_port_ptf_id_list, out_port_la
             }
         )
 
+    return setup_information
+
+def get_t2_duthost(duthosts, tbinfo):
+    """
+    Generate setup information dictionary for T2 topologies.
+    """
+    t3_duthost = find_duthost_on_role(duthosts, "T3", tbinfo)
+    t1_duthost = find_duthost_on_role(duthosts, "T1", tbinfo)
+    return t1_duthost, t3_duthost
+
+
+@pytest.fixture(scope="module")
+def setup_info(duthosts, rand_one_dut_hostname, tbinfo, request):
+    """
+    Gather all required test information.
+
+    Args:
+        duthost: DUT fixture
+        tbinfo: tbinfo fixture
+
+    Returns:
+        dict: Required test information
+
+    """
+    topo = tbinfo['topo']['name']
+    if 't1' in topo or 't0' in topo:
+        downstream_duthost = upstream_duthost = duthost = duthosts[rand_one_dut_hostname]
+    elif 't2' in topo:
+        downstream_duthost, upstream_duthost = get_t2_duthost(duthosts, tbinfo)
+        
+    setup_information = gen_setup_information(downstream_duthost, upstream_duthost, tbinfo)
+
     # Disable BGP so that we don't keep on bouncing back mirror packets
     # If we send TTL=1 packet we don't need this but in multi-asic TTL > 1
-    duthost.command("sudo config bgp shutdown all")
+
+    if 't2' in topo:
+        for duthost in duthosts.frontend_nodes:
+            duthost.command("sudo config bgp shutdown all")
+            duthost.command("mkdir -p {}".format(DUT_RUN_DIR))
```

Comment:
Is it necessary to put `duthost.command("mkdir -p {}".format(DUT_RUN_DIR))` in the for loop?

Reply:
yes, we need to create DUT_RUN_DIR on both everflow dut and remote dut. We send traffic on both dut to test mirroring happening on same dut and also across dut.